sql/inspect: skip REFCURSOR stored columns in consistency check

spilchen opened the pull request:
REFCURSOR values cannot be compared for equality, which breaks the `INSPECT` command when an index stores a column of this type. During a consistency check, INSPECT attempts to compare stored column payloads between the primary and secondary indexes. This fails for REFCURSOR values because the column type cannot be used in an equality comparison.

This change skips REFCURSOR stored columns during the primary/secondary comparison.

Informs: #155676
Epic: CRDB-55075

Release note (bug fix): INSPECT can now be run on tables with indexes that store REFCURSOR-typed columns.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/155772)
<!-- Reviewable:end -->

fqazi reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

One minor suggestion

> @fqazi reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/155772#-:-OcBIwCC0NeBI52K8eQM:b-g82hfq)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @spilchen)

___
*[`pkg/sql/inspect/index_consistency_check.go` line 126 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/155772#-OcBIl6ECOen6CqkUS0f:-OcBIl6ECOen6CqkUS0g:b4p8es8) ([raw file](https://github.com/cockroachdb/cockroach/blob/ae84d633e1a90b10fc6edf66e68895b0611e9b2f/pkg/sql/inspect/index_consistency_check.go#L126)):*
> ```Go
> 		col := c.tableDesc.PublicColumns()[pos]
> 		otherColumns = append(otherColumns, col)
> 		if col.GetType().Family() != types.RefCursorFamily {
> ```

Nit: Lets just have it return for refcursor instead?


<!-- Sent from Reviewable.io -->

spilchen committed:
sql/inspect: skip REFCURSOR stored columns in consistency check

REFCURSOR values cannot be compared for equality, which breaks the
`INSPECT` command when an index stores a column of this type. During a
consistency check, INSPECT attempts to compare stored column payloads
between the primary and secondary indexes. This fails for REFCURSOR
values because the column type cannot be used in an equality comparison.

This change skips REFCURSOR stored columns during the primary/secondary
comparison.

Informs: #155676
Epic: CRDB-55075

Release note (bug fix): INSPECT can now be run on tables with indexes
that store REFCURSOR-typed columns.

spilchen reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/155772)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @fqazi)

___
*[`pkg/sql/inspect/index_consistency_check.go` line 126 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/155772#-OcBIl6ECOen6CqkUS0f:-OcBjIrR2fkHWzyP-UVM:bb6un1j) ([raw file](https://github.com/cockroachdb/cockroach/blob/ae84d633e1a90b10fc6edf66e68895b0611e9b2f/pkg/sql/inspect/index_consistency_check.go#L126)):*
<details><summary><i>Previously, fqazi (Faizan Qazi) wrote…</i></summary><blockquote>

Nit: Lets just have it return for refcursor instead?
</blockquote></details>

Done


<!-- Sent from Reviewable.io -->
